Add elixir_make workaround

codyps opened the pull request:
If we don't set ELIXIR_MAKE_CACHE_DIR, the elixir_make package attempts to write to a cache directory under $HOME, which fails.

Get many of the elixir_make items working by adding this and adding gnumake as a build input.

For some elixir_make packages, they'll still need additional depenencies, which can be added on a package name basis.

Note: It feels like this and rustlerPrecompiled, which both examine dependencies to decide on workarounds could potentially be done purely in nix instead of examining the dependencies in the deps_nix code. Doing that would also make it more clear to others how to add local workarounds for all crates that depend on a particular crate. I've kept this in deps_nix's elixir code for now for consistency with rustlerPrecompiled.

Note also: this doesn't apply both workarounds if a package depends on both rustler_precompiled and elixir_make, it prefers applying the rustlerPrecompiled workaround if rustler_precompiled is depended on. We _could_ apply the workarounds in series, but I'm not sure that would make sense, and in practice packages tend to either depend on rustler_precompiled or elixir_make and not on both.

camelpunch commented:
I hear you on the ease-of-contribution issue: I think a restructuring to make it more obvious how to add a package that's expected to work could help here. I'll think about it.

codyps commented:
Thanks for pointing me in the right direction here. I've added the changes to ensure exqlite builds, and fixed the breakage to the vix test due to adding this (we may want to consider if we might want to tweak the override generation so it doesn't apply to the vix special case)

tjarratt commented:
Having a look at this, I think I see what happened. When we apply the `elixir_make` workaround for `lazy_html`, that applies to `vix` (because it depends upon `elixir_make`), and since [our override sets the `nativeBuildInputs`](https://github.com/code-supply/deps_nix/pull/22/files#diff-1c75c0e2f179e7b8d793703c329381499707c49238c9c0e686d625fad0f71e12R19)  for the derivation, the [ones we had previously set for vix](https://github.com/code-supply/deps_nix/pull/22/files#diff-1c75c0e2f179e7b8d793703c329381499707c49238c9c0e686d625fad0f71e12L1510) end up getting overriden.

I think I see a few ways to handle this, I'll take a pass at it.
